users/users/users.php

```diff
@@ -45,6 +45,115 @@ public function delete()
 	 *
 	 * @return void
 	 */
+	 
+	 /**
+	 * Function for edit user record.
+	 *
+	 * @return void
+	 */
+	public function put()
+	{
+		$app              = JFactory::getApplication();
+		$db 			  = JFactory::getDbo();
+		$data             = array();
+		$eobj             = new stdClass();	
+		
+		// Get user details to update
+		$data['username']      = $app->input->get('username', '', 'STRING');
+		$data['user_id']       = $app->input->get('user_id', '', 'INT');
+		$data['email']         = $app->input->get('email', '', 'STRING');
+		$data['block']         = $app->input->get('block', '', 'INT');
+		$data['name']     	   = $app->input->get('name', '', 'STRING');
+		$data['sendEmail']     = $app->input->get('sendEmail', '', 'INT');
+		$data['password']      = $app->input->get('password', '', 'STRING');
+	
+		// Check username or user_id to edit the details of user
+		if (!empty($data['username']) || !empty($data['user_id']))
+		{
+			if (empty($data['user_id']) && !empty($data['username']))
+			{
+				// Get user_id with the help of username
+				$query = $db->getQuery(true);
+				$query->select('id');
+				$query->from('#__users');
+				$query->where($db->quoteName('username') . '=' .  $db->quote($data['username']));
+				$db->setQuery($query);
+				$user_id = $db->loadResult();
+				$data['user_id'] = $user_id;
+			}
+			
+				$user = JFactory::getUser($data['user_id']);
+				
+				if (!isset($user->id))
+				{
+					// Not given username or user_id to edit the details of user
+					$eobj->status = false;
+					$eobj->id = 0;
+					$eobj->code = '404';
+					$eobj->message = JText::_( 'PLG_API_USERS_REQUIRED_USER_NOTEXIST_MESSAGE' );	
+					$this->plugin->setResponse($eobj);
+					return;
+				}
+				
+				if (!empty($data['block'] ))
```

Comment:
Okay...Will do changes  and let you know.

On Tue, Aug 23, 2016 at 4:28 PM, Ashwin Date <notifications@github.com>
wrote:

> In users/users/users.php
> <https://github.com/techjoomla/com_api-plugins/pull/36#discussion_r75843615>
> :
>
> > +			}
> > +			
> > +				$user = JFactory::getUser($data['user_id']);
> > +				
> > +				if (!isset($user->id))
> > +				{
> > +					// Not given username or user_id to edit the details of user
> > +					$eobj->status = false;
> > +					$eobj->id = 0;
> > +					$eobj->code = '404';
> > +					$eobj->message = JText::_( 'PLG_API_USERS_REQUIRED_USER_NOTEXIST_MESSAGE' );	
> > +					$this->plugin->setResponse($eobj);
> > +					return;
> > +				}
> > +				
> > +				if (!empty($data['block'] ))
>
> Check the user model. You should use the bind() method instead of doing
> this for each field.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/techjoomla/com_api-plugins/pull/36/files/f2b1954e35dacfe59d0a2b46a50323bc07404737#r75843615>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUNb4llV3mjWg0MNa8nHjguF7p1Oxc5Tks5qitJYgaJpZM4JqyuW>
> .
>



-- 
Best Regards,
Deepali A. Madane